runtime/list.go

```diff
@@ -291,6 +300,29 @@ func listNE(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return listCompare(f, toListUnsafe(v), w, NE)
 }
 
+func listPop(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	argc := len(args)
+	expectedTypes := []*Type{ListType, IntType}
+	if argc == 1 {
+		expectedTypes = expectedTypes[:1]
+	}
+	if raised := checkMethodArgs(f, "pop", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	i := -1
+	if argc == 2 {
+		i = toIntUnsafe(args[1]).Value()
+	}
+	l := toListUnsafe(args[0])
+	if i < 0 {
+		i += len(l.elems)
+	}
+	if i >= len(l.elems) || i < 0 {
+		return nil, f.RaiseType(IndexErrorType, "list index out of range")
+	}
+	return l.Pop(i), nil
```

Comment:
You need to wrap the length validation in the mutex, since otherwise concurrent calls on the same object could race. I suggest removing the `(*List) Pop` method and just do the whole mutation in listPop.